consensus/src/chained_bft/event_processor.rs

```diff
@@ -402,17 +402,18 @@ impl<T: Payload> EventProcessor<T> {
             Some((vote, vote_round)) if (*vote_round == round) => Some(vote.clone()),
             _ => {
                 // Try to generate a backup vote
-                match self.gen_backup_vote(round).await {
-                    Ok(backup_vote_msg) => {
+                self.gen_backup_vote(round)
+                    .await
+                    .map(|backup_vote_msg| {
                         self.last_vote_sent
                             .replace((backup_vote_msg.clone(), round));
-                        Some(backup_vote_msg)
-                    }
-                    Err(e) => {
+                        backup_vote_msg
+                    })
+                    .map_err(|e| {
                         warn!("Failed to generate a backup vote: {}", e);
-                        None
-                    }
-                }
+                        e
+                    })
+                    .ok()
```

Comment:
I've noticed a lot of these across the codebase. But what's wrong with a `match`? I actually find `match` clearer to read.

Reply:
The usages of `.ok()` I could find across the code base are unrelated to what's going on in this particular instance.

Reply:
I actually prefer the match as well and find it clearer.

Reply:
The construction suggested here breaks down the logging/last vote update and the conversion to a `Result` in separate places. But happy to remove a71453dfb9c8cf113a182f779a2527f547c217b7 if you want.

Reply:
Let remove it since it's somewhat grey.

Reply:
( @huitseeker I meant using .map() and .map_err() instead of a match )

Reply:
@aching refactored as a71453dfb9c8cf113a182f779a2527f547c217b7 upon @mimoo 's advice